uwblueprint/humane-society
# Created model and migration files for behaviour and behaviour level details

## Notion ticket link
[Ticket Name](https://www.notion.so/uwblueprintexecs/Extend-Behaviours-Entity-I-10d10f3fb1dc80988716e2865f807120?pvs=4)



## Implementation description
* Added a model for `behaviour_level_details` and a migration file with up and down functions
* Updated the `behaviour` model with `parent_behaviour_id` column and created a new migration file



## Steps to test
1. Run `node migration up`
2. The new table `behaviour_level_details` should be created
3. Run `node migrate down` twice
4. The new column(`parent_behaviour_id`) from `behaviours` should be removed and `behaviour_level_details` should be removed.


## What should reviewers focus on?
* The type for level is `INTEGER` instead of `ENUM` since enums cannot have integer values. The validation is instead done in the migration file for `behaviour_level_details` to check that the min value is 1 and max value is 4.


## Checklist
